pkg/licensing/scanner.go

```diff
@@ -52,3 +71,50 @@ func categoryToSeverity(category types.LicenseCategory) dbTypes.Severity {
 	}
 	return dbTypes.SeverityUnknown
 }
+
+func (s *Scanner) licenseToCategory(license expression.SimpleExpr) types.LicenseCategory {
+	for category, names := range s.categories {
+		if slices.Contains(names, license.License) {
+			return category
+		}
+	}
+	return types.CategoryUnknown
+}
+
+func (s *Scanner) compoundLicenseToCategory(license expression.CompoundExpr, visited map[string]types.LicenseCategory) types.LicenseCategory {
+	switch license.Conjunction() {
+	case expression.TokenAnd:
+		return s.compoundLogicOperator(license, visited, true)
+	case expression.TokenOR:
+		return s.compoundLogicOperator(license, visited, false)
+	default:
+		return types.CategoryUnknown
+	}
+}
+
+func (s *Scanner) compoundLogicOperator(license expression.CompoundExpr, visited map[string]types.LicenseCategory, findMax bool) types.LicenseCategory {
+	lCategory := s.traverseLicenseExpression(license.Left().String(), visited)
+	lSeverity := categoryToSeverity(lCategory)
+	rCategory := s.traverseLicenseExpression(license.Right().String(), visited)
+	rSeverity := categoryToSeverity(rCategory)
+
+	if lSeverity == dbTypes.SeverityUnknown || rSeverity == dbTypes.SeverityUnknown {
+		return types.CategoryUnknown
+	}
+
+	var logicOperator int
+	if findMax {
+		logicOperator = 1
+	} else {
+		logicOperator = -1
+	}
+
+	var compoundCategory types.LicenseCategory
+	if 0 < logicOperator*dbTypes.CompareSeverityString(lSeverity.String(), rSeverity.String()) {
+		compoundCategory = rCategory
+	} else {
+		compoundCategory = lCategory
+	}
+
+	return compoundCategory
```

Comment:
Reduced the number of lines of code a little :)
```suggestion
	// Compare the two severities:
	// - `cmp < 0` means left > right
	// - `cmp > 0` means left < right
	cmp := dbTypes.CompareSeverityString(lSeverity.String(), rSeverity.String())

	// For the maximum, pick the left side when left >right.
	// For the minimum, pick the left side when left < right.
	if (findMax && cmp < 0) || (!findMax && cmp > 0) {
		return lCategory
	}
	return rCategory
```